### ISSUE TITLE
checking of return value of evutil_snprintf

### ISSUE THREAD

**Issue by viennadd** [libevent/libevent#461]
In function `evutil_inet_ntop` of evutil.c [evutil.c#L1863](https://github.com/libevent/libevent/blob/master/evutil.c#L1863), while handling `if af == AF_INET`, there is a negative return value checking after calling `evutil_snprintf`, but not in af == AF_INET6, would it be an inconsistency? 

Regards

**Comment by azat** [comment]
I looked through the glibc's `snprintf()` implementation, and at first glance the only possible case when it there is not enough length in user-passed buffer, so I guess that this unlikely, but maybe it is worth checking.

**Comment by ploxiln** [comment]
If there is not enough space in the user-passed buffer, `snprintf()` returns the length that _would have_ been written if there was plenty of space. So it does not return negative in that case either.

The macOS snprintf() man page is more clear than the Linux one when it comes to errors.

> These functions return the number of characters printed (not including the trailing `\0` used to end output to strings), except for snprintf() and vsnprintf(), which return the number of characters that would have been printed if the size were unlimited (again, not including the final `\0`).  These functions return a negative value if an error occurs.

So what other errors could there be?

```
In addition to the errors documented for the write(2) system call, the printf() family of functions may fail if:

     [EILSEQ]           An invalid wide character code was encountered.

     [ENOMEM]           Insufficient storage space is available.
```

Errors documented for `write()` would only be possible for printf() family functions that actually write to a file descriptor, such as `printf()` or `fprintf()`. ENOMEM looks like it's only applicable to `asprintf()` (which allocates memory for you). EILSEQ is only possible if the `%lc` conversion is used, which I'm pretty sure libevent would never use.

In conclusion, I don't think `snprintf()` will ever return negative.

**Comment by viennadd** [comment]
Thank you @azat @ploxiln , so [evutil.c#L1877](https://github.com/libevent/libevent/blob/master/evutil.c#L1877) for checking `r < 0` should not be necessary then. since there is no negative check for `snprintf `in the `else if` block of `(af == AF_INET6)`
